DSharpPlus.Commands/ExtensionMethods.cs

```diff
@@ -0,0 +1,117 @@
+namespace DSharpPlus.Commands;
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+/// <summary>
+/// Extension methods used by the <see cref="CommandsExtension"/> for the <see cref="DiscordClient"/>.
+/// </summary>
+public static class ExtensionMethods
+{
+    /// <summary>
+    /// Registers the extension with the <see cref="DiscordClient"/>.
+    /// </summary>
+    /// <param name="client">The client to register the extension with.</param>
+    /// <param name="configuration">The configuration to use for the extension.</param>
+    public static CommandsExtension UseCommands(this DiscordClient client, CommandsConfiguration? configuration = null)
+    {
+        if (client is null)
+        {
+            throw new ArgumentNullException(nameof(client));
+        }
+        else if (client.GetExtension<CommandsExtension>() is not null)
+        {
+            throw new InvalidOperationException("Commands extension is already initialized.");
+        }
+
+        CommandsExtension extension = new(configuration ?? GrabDefaultConfiguration(client.Logger));
+        client.AddExtension(extension);
+        return extension;
+    }
+
+    /// <summary>
+    /// Registers the extension with all the shards on the <see cref="DiscordShardedClient"/>.
+    /// </summary>
+    /// <param name="shardedClient">The client to register the extension with.</param>
+    /// <param name="configuration">The configuration to use for the extension.</param>
+    public static async Task<IReadOnlyDictionary<int, CommandsExtension>> UseCommandsAsync(this DiscordShardedClient shardedClient, CommandsConfiguration? configuration = null)
+    {
+        ArgumentNullException.ThrowIfNull(shardedClient);
+
+        await shardedClient.InitializeShardsAsync();
```

Comment:
I dont think we should call this when initializing the extensions

Reply:
That's required to get the amount of shards/extensions needed for the sharded client

Reply:
I have mixed feelings because it overwirtes the internal discordClients and when the user uses ConnectAsync its called again

Reply:
If the method has already been called then it returns with the previous shard count. Additionally extensions/event handlers should be registered before the client starts, so I don't see any other way to avoid calling it twice.

Reply:
Oh i didnt saw the early return, all good